From bac4e2dc8d140b6425327f6ec2c23f739d67a18b Mon Sep 17 00:00:00 2001
From: Steev Klimaszewski <steev@kali.org>
Date: Thu, 15 Sep 2022 12:41:45 -0500
Subject: [PATCH] hack: drm/meson: venc: add support for WS7.9"

The Waveshare 7.9" display is 400x1280, so lets try a quick hack to make
it so it doesn't complain that 400 vertical is a bad mode

Signed-off-by: Steev Klimaszewski <steev@kali.org>
---
 drivers/gpu/drm/meson/meson_venc.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/meson/meson_venc.c b/drivers/gpu/drm/meson/meson_venc.c
index ebe4f2a92fe061..97f2bf05e249e5 100644
--- a/drivers/gpu/drm/meson/meson_venc.c
+++ b/drivers/gpu/drm/meson/meson_venc.c
@@ -866,11 +866,13 @@ meson_venc_hdmi_supported_mode(const struct drm_display_mode *mode)
 			    DRM_MODE_FLAG_PVSYNC | DRM_MODE_FLAG_NVSYNC))
 		return MODE_BAD;
 
-	/* support higher resolution than 1920x1080 */
-	if (mode->hdisplay < 640 || mode->hdisplay > 2560)
+	/* support 384x1280 and 1280x384 for the waveshare 7.9"
+	 * as well, support 2560x1600 for 2K resolution
+	 */
+	if (mode->hdisplay < 384 || mode->hdisplay > 2560)
 		return MODE_BAD_HVALUE;
 
-	if (mode->vdisplay < 480 || mode->vdisplay > 1600)
+	if (mode->vdisplay < 384 || mode->vdisplay > 1600)
 		return MODE_BAD_VVALUE;
 
 	return MODE_OK;
